### DISCUSSION TITLE
Complex Linetype Support (Batting)?

### DISCUSSION THREAD

**Discussion by mostl33** [mozman/ezdxf#1070] [Q&A] [answered]
Hi there!

First thanks for this great module! The featureset is kinda insane 👌 (in the good way) 😀.

Now a tiny question:

Is is possible to use "complex" linetypes with the SVG-Backend? Currently I looking for way to represent a insulation "pattern" in my SVG (e.g. "Batting"). 

Something like this:

<img width="587" alt="image" src="https://github.com/mozman/ezdxf/assets/73604048/ba283603-aed1-4817-8bbc-d584a05b69d4">

Best regards & and Thanks (!)
ml

**Comment by mozman** [comment, marked as answer]
No, complex linetypes are not supported by the `drawing` add-on.

From my experience as a professional CAD designer, such patterns often have a problem with non-rectangular ends and miters in general. This is particularly difficult to control in an automatically generated drawing.

**Comment by mozman** [comment]
Here's a script you can play around with. As you can see, solving the beginning and end of the pattern is not so easy:

```python
import ezdxf
from ezdxf.math import Vec2


def insulation_pattern(
    radius: float, height: float, count: int = 10
) -> list[tuple[Vec2, float]]:
    pattern: list[tuple[Vec2, float]] = []
    a = height / 2 - radius
    pen = Vec2(radius, -a)
    step1 = Vec2(-radius, 2 * a)
    step2 = Vec2(2 * radius, 0.0)
    step3 = Vec2(-radius, -2 * a)

    for _ in range(count):
        pattern.append((pen, 0.0))
        pen += step1
        pattern.append((pen, -1.0))
        pen += step2
        pattern.append((pen, 0.0))
        pen += step3
        pattern.append((pen, 1.0))
        pen += step2
    return pattern


def main():
    doc = ezdxf.new()
    msp = doc.modelspace()
    msp.add_lwpolyline(insulation_pattern(radius=5, height=20, count=10), format="vb")  # format=(vector, bulge)
    doc.saveas("insulation.dxf")

if __name__ == "__main__":
    main()
```

![image](https://github.com/mozman/ezdxf/assets/4230966/d2773219-c56f-4cfc-9da2-590e577c9f62)

**Reply by mostl33** [reply]
Oh  I see.

Thanks a lot for the example 🙏. We're currently evaluating the usage of ezdxf for visualizing layer structures (=  a basic dxf to svg converter, not using complex geometries just rectangles, allowing architects to visually define layer structures using a based tool).

Thanks a lot for the library and the example code _!_

Best regards
ml
